src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -111,10 +111,12 @@ static List<Type> GetHandlerMessageTypes(List<Type> list)
                     {
                         var messageType = @interface.GetGenericArguments()[0];
                         foundMessageTypes.Add(messageType);
+
+                        foundMessageTypes.AddRange(messageType.Assembly.GetTypes().Where(t=> messageType.IsAssignableFrom(t)));
```

Comment:
this has serious impact on the startup time @andreasohlund and honestly doesn't make much sense because the inherited type could be in another assembly as well. Why don't you just get the base types and implemented interfaces?